core/src/main/java/org/apache/carbondata/core/cache/CarbonLRUCache.java

```diff
@@ -95,9 +98,17 @@ public CarbonLRUCache(String propertyName, String defaultPropertyName) {
    * initialize lru cache
    */
   private void initCache() {
+    // get duration for cache expiration if configured
+    long duration = CarbonCommonConstants.CARBON_LRU_CACHE_EXPIRATION_DURATION_IN_MINUTES_DEFAULT;
+    String timeBasedExpiration = CarbonProperties.getInstance()
+        .getProperty(CarbonCommonConstants.CARBON_LRU_CACHE_EXPIRATION_DURATION_IN_MINUTES);
+    if (null != timeBasedExpiration) {
+      duration = Long.parseLong(timeBasedExpiration);
+    }
+    // initialise guava cache with time based expiration
     lruCacheMap =
-        new LinkedHashMap<String, Cacheable>(CarbonCommonConstants.DEFAULT_COLLECTION_SIZE, 1.0f,
-            true);
+        CacheBuilder.newBuilder().initialCapacity(CarbonCommonConstants.DEFAULT_COLLECTION_SIZE)
+            .expireAfterWrite(duration, TimeUnit.MINUTES).build();
```

Comment:
expire after write or access?

besides, guava also support size based cache, will it be supported later?

Reply:
Size-based eviction can be supported later if needed